Model/Product/Builder.php

```diff
@@ -135,6 +142,42 @@ public function build(Product $product, StoreInterface $store)
     }
 
     /**
+     * Helper method to fetch and return the normalised rating value for a product. The rating is
+     * normalised to a 0-5 value.
+     *
+     * @param Product $product the product whose rating value to fetch
+     * @param StoreInterface $store the store scope in which to fetch the rating
+     * @return float the normalized rating value of the product
+     */
+    private function buildRatingValue(Product $product, StoreInterface $store) {
+        /** @noinspection PhpUndefinedMethodInspection */
+        if (!$product->getRatingSummary()) {
+            $this->reviewFactory->create()->getEntitySummary($product, $store->getId());
```

Comment:
That's just the way you're supposed to do it. https://github.com/magento/magento2/blob/2.1.5/app/code/Magento/Review/Block/Product/ReviewRenderer.php#L72